monkestation/code/modules/clothing/shoes/magboots.dm

```diff
@@ -0,0 +1,71 @@
+/obj/item/clothing/shoes/magboots/boomboots
+
+	desc = "The ultimate in clown shoe technology."
+	name = "boom boots"
+	icon_state = "boomboot0"
+	item_state = "boomboot0"
+	magboot_state = "boomboot"
+	slowdown = SHOES_SLOWDOWN+1
+	item_color = "boomboots"
+	actions_types = list(/datum/action/item_action/toggle)
+	pocket_storage_component_path = /datum/component/storage/concrete/pockets/shoes/clown
+	var/datum/component/waddle
+	var/enabled_waddle = TRUE
+
+/obj/item/clothing/shoes/magboots/boomboots/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, list('monkestation/sound/misc/Boomboot1.ogg'=1), 50)
+
+/obj/item/clothing/shoes/magboots/boomboots/equipped(mob/user, slot)
+	. = ..()
+	if(slot == ITEM_SLOT_FEET)
+		if(enabled_waddle)
+			waddle = user.AddComponent(/datum/component/waddling)
+		if(user.mind && user.mind.assigned_role == "Clown")
+			SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "clownshoes", /datum/mood_event/clownshoes)
+
+/obj/item/clothing/shoes/magboots/boomboots/item_action_slot_check(slot, mob/user)
+	if(slot == ITEM_SLOT_FEET)
+		return 1
+
+/obj/item/clothing/shoes/magboots/boomboots/attack_hand(mob/user)
+	if(iscarbon(user))
+		var/mob/living/carbon/C = user
+		if(src == C.shoes && magpulse)
+			to_chat(user, "<span class='userdanger'>The boomboots anti-tamper system doesn't allow you to remove them while on!</span>")
+			return
+	..()
+
+/obj/item/clothing/shoes/magboots/boomboots/dropped(mob/user)
+	. = ..()
+	QDEL_NULL(waddle)
+	if(user.mind && user.mind.assigned_role == "Clown")
+		SEND_SIGNAL(user, COMSIG_CLEAR_MOOD_EVENT, "clownshoes")
+	if(magpulse)//make sure they're being worn and activated
+		explosion(src,2,4,8,6)//used the size of the big rubber ducky bomb
+
+/obj/item/clothing/shoes/magboots/boomboots/attack_self(mob/user)
+	if(iscarbon(user))
+		var/mob/living/carbon/C = user
+		if(src == C.shoes)
+			magpulse = !magpulse
+			if(magpulse)
+				clothing_flags &= ~NOSLIP
+				strip_delay = 100
+			else
+				clothing_flags |= NOSLIP
```

Comment:
use the add flag and remove flag macros for readability. Can't remember exactly what they're called

Reply:
replacing the lines with ADD_TRAIT and REMOVE_TRAIT made them function improperly so just gonna leave it there.  This is consistent with existing magboots anyways and since it's a child of it, seems weird to change it.

Reply:
I dont think its add trait, its something like add flag